XTS/WS-T/dev/src/org/jboss/jbossts/xts/recovery/participant/at/XTSATRecoveryManager.java

```diff
@@ -166,4 +191,6 @@ public static XTSATRecoveryManager setRecoveryManager(XTSATRecoveryManager recov
      */
 
     protected static XTSATRecoveryManager theRecoveryManager = null;
+    protected final static Object lock = new Object();
+    private final static int MAX_INIT_TIMEOUT_MILLIS = 10000;
```

Comment:
I think we should just wait with no particular timeout, presumably when setRecoveryManager is not called that is some symptom of a significant misconfiguration?

Reply:
If there is a good reason to not wait without timeout then I think some of the callers of this method need to be made more able to deal with null: https://github.com/jbosstm/narayana/search?p=1&q=getRecoveryManager

Reply:
The server will hang - I thought that failing with a exception would be kinder.
But I agree that not setting it or setting it to zero would be an XTS implementation error (assuming that the application cannot set the recovery manager).